reward/stakingInfo.go

```diff
@@ -0,0 +1,154 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package reward
+
+import (
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/params"
+	"math"
+	"math/big"
+	"sort"
+)
+
+const (
+	AddrNotFoundInCouncilNodes = -1
+	maxStakingLimit            = uint64(100000000000)
+	DefaultGiniCoefficient     = -1.0
+)
+
+var (
+	maxStakingLimitBigInt = big.NewInt(0).SetUint64(maxStakingLimit)
+)
+
+// StakingInfo contains staking information.
+type StakingInfo struct {
+	BlockNum uint64 // Block number where staking information of Council is fetched
+
+	// Information retrieved from AddressBook smart contract
+	CouncilNodeIds       []common.Address // NodeIds of Council
+	CouncilStakingdAddrs []common.Address // Address of Staking account which holds staking balance
+	CouncilRewardAddrs   []common.Address // Address of Council account which will get block reward
+	KIRAddr              common.Address   // Address of KIR contract
+	PoCAddr              common.Address   // Address of PoC contract
+
+	UseGini bool
+	Gini    float64 // gini coefficient
+
+	// Derived from CouncilStakingAddrs
+	CouncilStakingAmounts []uint64 // Staking amounts of Council
+}
+
+func newEmptyStakingInfo(blockNum uint64) (*StakingInfo, error) {
+	stakingInfo := &StakingInfo{
+		BlockNum:              blockNum,
+		CouncilNodeIds:        make([]common.Address, 0, 0),
+		CouncilStakingdAddrs:  make([]common.Address, 0, 0),
+		CouncilRewardAddrs:    make([]common.Address, 0, 0),
+		KIRAddr:               common.Address{},
+		PoCAddr:               common.Address{},
+		CouncilStakingAmounts: make([]uint64, 0, 0),
+		Gini:                  DefaultGiniCoefficient,
+		UseGini:               false,
+	}
+	return stakingInfo, nil
+}
+
+func newStakingInfo(bc *blockchain.BlockChain, helper governanceHelper, blockNum uint64, nodeIds []common.Address, stakingAddrs []common.Address, rewardAddrs []common.Address, KIRAddr common.Address, PoCAddr common.Address) (*StakingInfo, error) {
+	intervalBlock := bc.GetBlockByNumber(blockNum)
+	statedb, err := bc.StateAt(intervalBlock.Root())
+	if err != nil {
+		logger.Trace("Failed to make a state for interval block", "interval blockNum", blockNum, "err", err)
+		return nil, err
+	}
+
+	// Get balance of stakingAddrs
+	var stakingAmounts []uint64
+	stakingAmounts = make([]uint64, len(stakingAddrs))
```

Comment:
How about `stakingAmounts := make([]uint64, len(stakingAddrs))`

Reply:
I've reflected it